includes/API.php

```
@@ -529,7 +529,16 @@ private function update_order_with_payment_intent( $payment_intent ): void {
 			return;
 		}
 
-		// Save payment metadata instead of completing the order immediately.
+		// A stale or unrelated intent must not overwrite or complete the order:
+		// when a Terminal attempt is recorded, only that intent is accepted.
+		$recorded_intent = (string) $order->get_meta( '_stripe_terminal_payment_intent_id' );
+		if ( '' !== $recorded_intent && $recorded_intent !== $payment_intent->id ) {
+			Logger::log( 'Payment intent webhook: ignoring intent ' . $payment_intent->id . ' for order ' . $order_id . '; the recorded Terminal intent is ' . $recorded_intent, 'warning' );
```

PHPStan (2.2.8)

[error] Static method WCPOS\WooCommercePOS\StripeTerminal\Logger::log() invoked with 2 parameters, 1 required.


(arguments.count)